### PR TITLE
add Requirement from garbo

### PR DESCRIPTION
**Pull request by horrible-little-slime**

### CODE REVIEW

**Comment by gausie** [comment]
Use a leading `_` instead of a trailing `_` for the private values, or better yet use the [new private syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields). Then you could maybe use a getter for the public method.

File: src/maximize.ts [hunk 1]
```diff
@@ -329,3 +329,65 @@ export function maximizeCached(
...
+    return this.maximizeOptions_;
+  }
+
+  merge(other: Requirement): Requirement {
+    const optionsA = this.maximizeOptions();
+    const optionsB = other.maximizeOptions();
+    return new Requirement(
+      [...this.maximizeParameters(), ...other.maximizeParameters()],
+      {
+        ...optionsA,
```
Comment (gausie) [inline src/maximize.ts:359]: Can we not just use deepMerge from lodash here?

Comment (horrible-little-slime) [reply src/maximize.ts]: Changed; I think I've done what you're looking for, but I'm not 100% sure.

File: src/maximize.ts [hunk 2]
```diff
@@ -329,3 +330,40 @@ export function maximizeCached(
...
+export class Requirement {
+  #maximizeParameters: string[];
+  #maximizeOptions: MaximizeOptions;
+
+  constructor(maximizeParameters: string[], maximizeOptions: MaximizeOptions) {
+    this.#maximizeParameters = maximizeParameters;
+    this.#maximizeOptions = maximizeOptions;
+  }
+
+  maximizeParameters(): string[] {
```
Comment (gausie) [inline src/maximize.ts:343]:
```suggestion
  get maximizeParameters(): string[] {
```

File: src/maximize.ts [hunk 3]
```diff
@@ -329,3 +330,40 @@ export function maximizeCached(
...
+    return this.#maximizeParameters;
+  }
+
+  maximizeOptions(): MaximizeOptions {
```
Comment (gausie) [inline src/maximize.ts:347]:
```suggestion
  get maximizeOptions(): MaximizeOptions {
```

File: src/maximize.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ import {
 } from "kolmafia";
 import { $familiar, $item, $slot, $slots, $stats } from "./template-string";
 import logger from "./logger";
+import _ from "lodash";
```
Comment (gausie) [inline src/maximize.ts:17]:
```suggestion
import merge from "lodash/merge";
```

File: src/maximize.ts [hunk 5]
```diff
@@ -329,3 +330,40 @@ export function maximizeCached(
...
+  maximizeParameters(): string[] {
+    return this.#maximizeParameters;
+  }
+
+  maximizeOptions(): MaximizeOptions {
+    return this.#maximizeOptions;
+  }
+
+  merge(other: Requirement): Requirement {
+    return _.merge(this, other);
```
Comment (gausie) [inline src/maximize.ts:352]:
```suggestion
    return merge(this, other);
```

**Review by gausie — CHANGES_REQUESTED** [review]

**Pull request merged**